st2actions/st2actions/container/base.py

```diff
@@ -160,8 +164,31 @@ def _do_run(self, runner, runnertype_db, action_db, liveaction_db):
         runner.post_run(status, result)
         runner.container_service = None
 
+        LOG.debug('Runner do_run result', extra={'result': updated_liveaction_db.result})
+        LOG.audit('Liveaction completed', extra={'liveaction_db': updated_liveaction_db})
+
         return updated_liveaction_db
 
+    def _do_cancel(self, runner, runnertype_db, action_db, liveaction_db):
+        try:
+            extra = {'runner': runner}
+            LOG.debug('Performing cancel for runner: %s', (runner.runner_id), extra=extra)
+
+            runner.cancel()
+
+            liveaction_db.status = action_constants.LIVEACTION_STATUS_CANCELED
+            liveaction_db.end_timestamp = date_utils.get_datetime_utc_now()
+            liveaction_db.result = {'message': 'Action canceled by user.'}
+            liveaction_db = LiveAction.add_or_update(liveaction_db)
+            executions.update_execution(liveaction_db)
```

Comment:
So in this case, you are writing the message. So you might be fine. Still remember, add_or_update could throw.